### DISCUSSION TITLE
Is it possible to make a BE/Server/Node Controller in Next.js?

### DISCUSSION THREAD

**Discussion by OZZlE** [vercel/next.js#21986] [Help] [answered]
If I for example want to create a Server-side (Node) endpoint that only returns json. This server endpoint has perhaps ssh keys and other sensitive authentication that we don't want to leak clientside, but it also handles that only certain data is sent in and returned from it. The endpoint creates own requests to other systems.

In short:
Runs: server-side/node
Returns: only json (no html)
Used from clientside Next.js

This is possible using Express.Js to host Next.js for example but if one doesn't want to use Express.js and host it directly on Vercel or other Next.js hosting providers.

I tried a bit with something like getServerSideProps() on a /page/ but it returns the header, footer, js, css and the page layout, when I only want json returned.

It would be really useful I think to be able to do this without setting up Express.Js and having special hosting needs due to that.

**Comment by jamesmosier** [comment, marked as answer]
Hi @OZZlE. It sounds like [API routes](https://nextjs.org/docs/api-routes/introduction) are exactly what you are mentioning! These can return JSON and are kept private from the client side. 

Also `getServerSideProps` would work as well, as this returns `props` to your components when the page loads. There is no way to explicitly call this though, as it is fired automatically when you navigate to a page.
